File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];
        HashMap<Integer, Integer> hM = new HashMap<>();
        for(int i=0; i<nums.length; i++){
            int remainder = target - nums[i];
            if(!hM.containsKey(remainder)){
                hM.put(nums[i], i);
            }
            else{
                res[0] = hM.get(remainder);
                res[1] = i;
                return res;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:

        def backtrack(start, curr_total, current):
            if curr_total == target:
                res.append(list(current))
                return
            for i in range(start, len(nums)):
                if curr_total + nums[i] > target:
                    continue
                current.append(nums[i])
                backtrack(i, curr_total + nums[i], current)
                current.pop()

        res = []
        nums.sort()
        backtrack(0, 0, [])

        return res

        
